ksql-engine/src/main/java/io/confluent/ksql/function/udf/datetime/StringToTimestamp.java

```diff
@@ -16,35 +16,38 @@
 
 package io.confluent.ksql.function.udf.datetime;
 
-import java.text.DateFormat;
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 
 import io.confluent.ksql.function.KsqlFunctionException;
 import io.confluent.ksql.function.udf.Kudf;
 
 public class StringToTimestamp implements Kudf {
 
-  private DateFormat dateFormat = null;
+  private DateTimeFormatter formatter;
```

Comment:
this could be initialized by more than 1 thread - right? Should we at least make it `volatile`? Or perhaps we need to synchronize the initialization?

Reply:
Thought about that, but don't see it as an issue.  

Without `volatile`, the worst that will happen is that all threads will create a new, identical, formatter instance when they first run. That's no big impact on performance.

Adding `volatile` would mean a memory barrier on each read of the variable. Which would adversely affect performance.

Reply:
To be clear, the formatter is guaranteed to be thread safe? If so, would be good to add a code comment about this.

Reply:
I have to be honest, I'm not a fan of such comments. If a reader wants to know if `DateTimeFormatter` is thread-safe, then IMHO they should read the docs for `DateTimeFormatter`, not rely on on secondary information in other comments...

What do others think?

Reply:
I've renamed `formatter` to `threadSafeFormatter`...

Reply:
i agree with Andy